Source/objects.cpp

```diff
@@ -1437,34 +1445,34 @@ void ActivateTrapLine(int ttype, int tid)
 void UpdateFlameTrap(Object &trap)
 {
 	if (trap._oVar2 != 0) {
-		if (trap._oVar4 != 0) {
+		if (trap._oVar4 != DOOR_CLOSED) {

```

Comment:
Same here, this function operates on a flamehole object, not a door.

Reply:
Oops, lets try that again, but this time with a bit of clean up on top :D